src/data/hash_map.lean

```diff
@@ -227,13 +227,13 @@ theorem valid.replace_aux (a : α) (b : β a) : Π (l : list (Σ a, β a)), a 
 | (⟨a', b'⟩::t) := begin
   by_cases e : a' = a,
   { subst a',
-    suffices : ∃ u w (b'' : β a),
-      sigma.mk a b' :: t = u ++ ⟨a, b''⟩ :: w ∧
+    suffices : ∃ (u w : list Σ a, β a) (b'' : β a),
+      list.cons (sigma.mk a b') t = u ++ ⟨a, b''⟩ :: w ∧
```

Comment:
Hmmm... I think it's sad that we would have to make these changes. Does it help if you write `(sigma.mk a b' : list) :: t` instead?

Reply:
That syntax does not work.  You can do `((sigma.mk a b') :: t : (list _))` or `[sigma.mk a b'] ++ t`, which might be better.  Or we could try to work out whether `multiset.cons` and `vector.cons` really need to be in scope anyway.

Reply:
I reorganised the pnat files in a way that made the changes to hash_map.lean unneccessary, so I removed them.